### PR TITLE
DEPR: enforce deprecation on floats with unit=M or Y

### PR DESCRIPTION
**Pull request by jbrockmendel**
Unfortunately just found that we missed the analogous behavior in to_datetime with non-scalars.  That'll go in the just-do-in-2.0 pile.

Whatsnew is in 1.6 on the assumption it will be changed to 2.0.

### CODE REVIEW

File: doc/source/whatsnew/v1.6.0.rst [hunk 1]
```diff
@@ -116,6 +116,7 @@ See :ref:`install.dependencies` and :ref:`install.optional_dependencies` for mor
 
 Other API changes
 ^^^^^^^^^^^^^^^^^
+- Disallow passing non-round floats to :class:`Timestamp` with ``unit="M"`` or ``unit="Y"`` (:issue:`47266`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.6.0.rst:119]: Could you start a new section in the whatsnew for these like we did in 1.0? https://pandas.pydata.org/docs/whatsnew/v1.0.0.html#removal-of-prior-version-deprecations-changes

Comment (jbrockmendel) [reply doc/source/whatsnew/v1.6.0.rst]: updated + green

Comment (jbrockmendel) [reply doc/source/whatsnew/v1.6.0.rst]: after this is merged ill update the other deprecation-enforcing PRs to move the notes to the appropriate section

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @jbrockmendel

**Pull request merged**